### PR TITLE
DM-32467: Allow a Path object to be used in Butler constructor

### PR DESCRIPTION
**Pull request by timj**
## Checklist

- [ ] ran Jenkins
- [ ] added a release note for user-visible changes to `doc/changes`

### CODE REVIEW

File: tests/test_uri.py [hunk 1]
```diff
@@ -63,6 +64,10 @@ def testFile(self):
         self.assertFalse(uri.exists(), f"{uri} should not exist")
         self.assertEqual(uri.ospath, file)
 
+        path = pathlib.Path(file)
+        uri = ButlerURI(path)
+        self.assertEqual(uri.ospath, str(path))
```
Comment (mfisherlevine) [inline tests/test_uri.py]: Not sure what the creation of the `BtlerURI`, but just wanted to check that this is still OK with/without trailing slashes. If anything the test is more fragile, so it's clearly fine, but just thought I'd flag that potential difference, depending on the value of path in future.

**Review by mfisherlevine — APPROVED** [review]
Looks great.

File: tests/test_uri.py (hunk 1 above)
Comment (timj) [reply tests/test_uri.py]: ButlerURI is what is used as soon as anything arrives inside butler. It's the class that handles the abstraction of local file vs remote object store.

Comment (timj) [reply tests/test_uri.py]: I'm not sure I understand what you mean about fragility. I do see that I should be comparing `uri.ospath` to `file` so I will fix that.

Comment (mfisherlevine) [reply tests/test_uri.py]: It's fine, I just didn't know exactly how equality was being done internally, so wanted to make sure this was trailing-slash safe (though I'd have been surprised if it wasn't). All good.

Comment (timj) [reply tests/test_uri.py]: `ButlerURI` is very careful about trailing slash handling.

**Pull request merged**